lib/librdkafka/NProducer.js

```diff
@@ -272,7 +272,7 @@ class NProducer extends EventEmitter {
       throw new Error("partitionCount must be number.");
     }
 
-    return this._murmur(key) % partitionCount;
+    this._murmur(key, partitionCount);
```

Comment:
I believe a little bug was introduced in this release - previously the function returned the murmur result, but now it only calls the murmur function and always returns `undefined`. As a result, manual partition key cannot be provided by the application and a random partition is selected.

Reply:
@elizehavi you are right, good find! Bad review on my part back then. I have fixed the issue and released sinek@6.22.2